### PR TITLE
Added generamba support

### PR DESCRIPTION
**Pull request by AlexYanski**

### CODE REVIEW

File: Bootstrap/Presentation/Empty/EmpryBuilderImpl.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ class EmpryBuilderImpl: EmptyBuilder {
         self.dependency = dependency
     }
 
-    func build() -> UIViewController {
+    func build() -> EmptyViewController {
```
Comment (pkondrashkov) [inline Bootstrap/Presentation/Empty/EmpryBuilderImpl.swift]: Lets leave it as UIViewController. The caller side needs not knowing about the actual type.

File: Templates/mvp_module/Code/contract.swift.liquid [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+protocol {{ module_info.name }}Dependency {
+    var parent: UINavigationController { get }
+}
+
+protocol {{ module_info.name }}Builder {
+    func build(listener: {{ module_info.name }}Listener) -> {{ module_info.name }}ViewController
+}
+
+protocol {{ module_info.name }}Listener: AnyObject {
+    func on{{ module_info.name }}Back()
```
Comment (pkondrashkov) [inline Templates/mvp_module/Code/contract.swift.liquid]: lets call this on***Finish

File: Templates/mvp_module/Code/contract.swift.liquid [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//  {{ module_info.project_name }}
+//
+//  Created by {{ developer.name }} on {{ date }}.
+//  Copyright © {{ year }} {{ developer.company }}. All rights reserved.
+//
+
+import UIKit
+
+protocol {{ module_info.name }}Dependency {
+    var parent: UINavigationController { get }
```
Comment (pkondrashkov) [inline Templates/mvp_module/Code/contract.swift.liquid]: parent should be UIViewController

File: Templates/mvp_module/Code/contract.swift.liquid [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+protocol {{ module_info.name }}Router: AnyObject {
+    func back()
+}
+
+protocol {{ module_info.name }}Interactor: AnyObject {}
+
+protocol {{ module_info.name }}Presenter: AnyObject {
+    func viewDidLoad()
+    func viewWillAppear()
+    func handleBackPressed()
```
Comment (pkondrashkov) [inline Templates/mvp_module/Code/contract.swift.liquid]: lets be more explicit: handleBackButtonPressed
Make sure to add left bar button action to call this method on presenter.

File: Templates/mvp_module/Code/builder.swift.liquid [hunk 5]
```diff
@@ -0,0 +1,32 @@
+//
```
Comment (pkondrashkov) [inline Templates/mvp_module/Code/builder.swift.liquid]: Lets remove all this headers.

File: Bootstrap/Presentation/Empty/EmpryBuilderImpl.swift (hunk 1 above)
Comment (AlexYanski) [reply Bootstrap/Presentation/Empty/EmpryBuilderImpl.swift]: Done

File: Templates/mvp_module/Code/contract.swift.liquid (hunk 2 above)
Comment (AlexYanski) [reply Templates/mvp_module/Code/contract.swift.liquid]: Done

File: Templates/mvp_module/Code/contract.swift.liquid (hunk 3 above)
Comment (AlexYanski) [reply Templates/mvp_module/Code/contract.swift.liquid]: Fixed, but I don't get how we will be using `NavigationScene` in this case, if we don't have a `UINavigationController` dependency?

File: Templates/mvp_module/Code/contract.swift.liquid (hunk 4 above)
Comment (AlexYanski) [reply Templates/mvp_module/Code/contract.swift.liquid]: done

File: Templates/mvp_module/Code/builder.swift.liquid (hunk 5 above)
Comment (AlexYanski) [reply Templates/mvp_module/Code/builder.swift.liquid]: Done

File: Templates/mvp_module/Code/contract.swift.liquid (hunk 3 above)
Comment (pkondrashkov) [reply Templates/mvp_module/Code/contract.swift.liquid]: UINavigationController has .navigationController property wich continas UINavigationController that is displaying current view controller. So we can just pass the view itself and when we need the UINavigationController we can use that property.

File: Bootstrap/Presentation/Empty/EmptyViewController.swift [hunk 6]
```diff
@@ -42,11 +42,6 @@ class EmptyViewController: UIViewController {
         logoImageView.translatesAutoresizingMaskIntoConstraints = false
         presenter.viewDidLoad()
     }
-
-    override func viewWillAppear(_ animated: Bool) {
```
Comment (pkondrashkov) [inline Bootstrap/Presentation/Empty/EmptyViewController.swift]: Why did you remove this method, it is actually pretty convinient. And we still have this method in the template

File: Bootstrap/Presentation/Root/RootContract.swift [hunk 7]
```diff
@@ -13,7 +13,7 @@ protocol RootDependency {
 }
 
 protocol RootBuilder: AnyObject {
-    func build() -> UIViewController
+    func build() -> RootViewController
```
Comment (pkondrashkov) [inline Bootstrap/Presentation/Root/RootContract.swift]: Why? this should be just UIViewController.

File: Templates/mvp_module/Code/contract.swift.liquid (hunk 3 above)
Comment (AlexYanski) [reply Templates/mvp_module/Code/contract.swift.liquid]: done

File: Bootstrap/Presentation/Empty/EmptyViewController.swift (hunk 6 above)
Comment (AlexYanski) [reply Bootstrap/Presentation/Empty/EmptyViewController.swift]: fixed

File: Bootstrap/Presentation/Root/RootContract.swift (hunk 7 above)
Comment (AlexYanski) [reply Bootstrap/Presentation/Root/RootContract.swift]: fixed
